lib/datadog/tracing/contrib/opensearch/patcher.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require_relative '../../metadata/ext'
+require_relative '../analytics'
+require_relative 'ext'
+require_relative '../ext'
+require_relative '../integration'
+require_relative '../patcher'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module OpenSearch
+        # Patcher enables patching of 'opensearch' module.
+        module Patcher
+          include Contrib::Patcher
+
+          module_function
+
+          def patch
+            require 'uri'
+            require 'json'
+            require_relative 'quantize'
+
+            ::OpenSearch::Transport::Client.prepend(Client)
+          end
+
+          # Patches OpenSearch::Transport::Client module
+          module Client
+            # rubocop:disable Metrics/MethodLength
+            def perform_request(method, path, params = {}, body = nil, headers = nil)
+              response = nil
+              Tracing.trace('opensearch.query', service: datadog_configuration[:service_name]) do |span|
+                begin
+                  # Set generic tags
+                  span.set_tag(Tracing::Metadata::Ext::TAG_COMPONENT, Ext::TAG_COMPONENT)
+                  span.set_tag(Tracing::Metadata::Ext::TAG_KIND, Tracing::Metadata::Ext::SpanKind::TAG_CLIENT)
+                  span.set_tag(Contrib::Ext::DB::TAG_SYSTEM, Ext::TAG_SYSTEM)
+
+                  # Set argument tags
+                  span.set_tag(OpenSearch::Ext::TAG_METHOD, method)
+                  span.set_tag(OpenSearch::Ext::TAG_PATH, path)
+
+                  tag_params(params, span)
+                  tag_body(body, span)
+
+                  # Parse url
+                  original_url = transport.get_connection.full_url(path, {})
+                  url = URI.parse(original_url)
+                  host = url.host
+                  port = url.port
+                  scheme = url.scheme
+                  # Set url.user to nil to remove sensitive information
```

Comment:
Can you specify what information is being removed here?

Reply:
Changed to "# Set url.user to nil to remove sensitive information (i.e. user's username and password)"!